sklearn/neighbors/_unsupervised.py

```diff
@@ -32,44 +32,62 @@ class NearestNeighbors(NeighborsBase, KNeighborsMixin,
         Note: fitting on sparse input will override the setting of
         this parameter, using brute force.
 
-    leaf_size : int, optional (default = 30)
+    leaf_size : int, default=30
         Leaf size passed to BallTree or KDTree.  This can affect the
         speed of the construction and query, as well as the memory
         required to store the tree.  The optimal value depends on the
         nature of the problem.
 
-    metric : string or callable, default 'minkowski'
-        the distance metric to use for the tree.  The default metric is
+    metric : str or callable, default='minkowski'
+        The distance metric to use for the tree.  The default metric is
         minkowski, and with p=2 is equivalent to the standard Euclidean
         metric. See the documentation of the DistanceMetric class for a
         list of available metrics.
         If metric is "precomputed", X is assumed to be a distance matrix and
         must be square during fit. X may be a :term:`Glossary <sparse graph>`,
         in which case only "nonzero" elements may be considered neighbors.
 
-    p : integer, optional (default = 2)
+    p : int, default=2
         Parameter for the Minkowski metric from
         sklearn.metrics.pairwise.pairwise_distances. When p = 1, this is
         equivalent to using manhattan_distance (l1), and euclidean_distance
         (l2) for p = 2. For arbitrary p, minkowski_distance (l_p) is used.
 
-    metric_params : dict, optional (default = None)
+    metric_params : dict, default=None
         Additional keyword arguments for the metric function.
 
-    n_jobs : int or None, optional (default=None)
+    n_jobs : int or None, default=None
         The number of parallel jobs to run for neighbors search.
         ``None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
         ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
         for more details.
 
+    **kwargs : dict, default=None
+        Additional keyword arguments to be passed to parent classes.
```

Comment:
These are passed to `NeigboursBase` I think. I'm not sure we actually need this parameter, as all the `NeigboursBase.__init__` parameters seems to be accounted for? 

@louishuynh Could you please double check it, and if it is the case, remove the added documentation about `**kwargs` added in this PR and open a separate issue about these not being necessary?

Reply:
Should this PR wait / depend on #15523 or are we happy to treat this independently?

Reply:
Agreed, this has been done - the separate issue is #15523

Reply:
Thanks @louishuynh, I responded there. Let's keep this parameter undocumented for now.